src/Shared/LayoutRenderers/AspNetRequestUrlRenderer.cs

```diff
@@ -19,42 +18,100 @@ namespace NLog.Web.LayoutRenderers
+        public AspNetRequestUrlProperty Properties { get; set; } = AspNetRequestUrlProperty.Default;
+
         /// <summary>
         /// To specify whether to include / exclude the Query string. Default is false.
         /// </summary>
-        public bool IncludeQueryString { get; set; }
+        [Obsolete("Please use the Properties flags enumeration instead")]
+        public bool IncludeQueryString 
+        {
+            get => HasPropertiesFlag(AspNetRequestUrlProperty.QueryString);
+            set => SetPropertiesFlag(AspNetRequestUrlProperty.QueryString, value);
+        }
 
         /// <summary>
         /// To specify whether to include / exclude the Port. Default is false.
         /// </summary>
-        public bool IncludePort { get; set; }
+        [Obsolete("Please use the Properties flags enumeration instead")]
+        public bool IncludePort
+        {
+            get => HasPropertiesFlag(AspNetRequestUrlProperty.Port);
+            set => SetPropertiesFlag(AspNetRequestUrlProperty.Port, value);
+        }
 
         /// <summary>
         /// To specify whether to exclude / include the host. Default is true.
         /// </summary>
-        public bool IncludeHost { get; set; } = true;
+        [Obsolete("Please use the Properties flags enumeration instead")]
+        public bool IncludeHost
+        {
+            get => HasPropertiesFlag(AspNetRequestUrlProperty.Host);
+            set => SetPropertiesFlag(AspNetRequestUrlProperty.Host, value);
+        }
 
         /// <summary>
         /// To specify whether to exclude / include the scheme. Default is true.
         /// </summary>
-        public bool IncludeScheme { get; set; } = true;
+        [Obsolete("Please use the Properties flags enumeration instead")]
+        public bool IncludeScheme
+        {
+            get => HasPropertiesFlag(AspNetRequestUrlProperty.Scheme);
+            set => SetPropertiesFlag(AspNetRequestUrlProperty.Scheme, value);
+        }
 
         /// <summary>
         /// To specify whether to exclude / include the url-path. Default is true.
         /// </summary>
-        public bool IncludePath { get; set; } = true;
+        [Obsolete("Please use the Properties flags enumeration instead")]
+        public bool IncludePath
+        {
+            get => HasPropertiesFlag(AspNetRequestUrlProperty.Path);
+            set => SetPropertiesFlag(AspNetRequestUrlProperty.Path, value);
+        }
+
+        /// <summary>
+        /// Helper to set the bit in the flags enum properly
+        /// </summary>
+        /// <param name="bit"></param>
+        /// <param name="flag"></param>
+        private void SetPropertiesFlag(AspNetRequestUrlProperty bit, bool flag)
+        {
+            if (flag)
+            {
+                Properties |= bit;
+            }
+            else
+            {
+                Properties &= ~bit;
+            }
+        }
+
+        /// <summary>
+        /// helper to get the bit in the flags enum
```

Comment:
Please avoid XML-docs for private methods. Prefer that method-name explains what they do, than using comments.